product_code/hyperledger/api/src/test/scala/ChaincodeConnectionObjectTests.scala

```diff
@@ -1,66 +1,89 @@
 
-import de.upb.cs.uc4.hyperledger.ConnectionManager
+import de.upb.cs.uc4.hyperledger.{ChaincodeQuickAccess, ConnectionManager}
 import de.upb.cs.uc4.hyperledger.traits.ChaincodeTrait
-import org.scalatest.FunSuite
+import org.scalatest.wordspec.AnyWordSpec
 
 import scala.util.{Success, Using}
 
-class ChaincodeConnectionObjectTests extends FunSuite {
+class ChaincodeConnectionObjectTests extends AnyWordSpec {
 
-  test("Check command getAllCourses") {
-    // setup connection
-    val chaincodeConnection = ConnectionManager.createConnection()
+  val connectionManager = ConnectionManager()
 
-    // perform action
-    try{
-      //retrieve result on query
-      val courses = chaincodeConnection.getAllCourses()
+  "A ChaincodeConnection" when {
+    "initialized" should {
+      "Allow for getCALlCourses" in {
+        // setup connection
+        val chaincodeConnection = connectionManager.createConnection()
 
-      // test result
-      assert(courses != null, "courses returned was null")
-    } finally {
-      // close connection
-      chaincodeConnection.close();
-    }
-  }
+        // perform action
+        try {
+          //retrieve result on query
+          val courses = chaincodeConnection.getAllCourses()
+
+          // test result
+          assert(courses != null, "courses returned was null")
+        } finally {
+          // close connection
+          chaincodeConnection.close();
+        }
+      }
 
-  test ("Check full walk-through"){
-    Using(ConnectionManager.createConnection()) { chaincodeConnection: ChaincodeTrait =>
-      // initial courses
-      val courses = chaincodeConnection.getAllCourses()
-      assert(courses != null, "Get All courses returned null")
-      println("Courses: " + courses)
+      "Allow a full walkthrough" in {
+        val testResult = Using(connectionManager.createConnection()) { chaincodeConnection: ChaincodeTrait =>
+          // initial courses
+          val courses = chaincodeConnection.getAllCourses()
+          assert(courses != null, "Get All courses returned null")

```

Comment:
We agreed to use Matchers with WordSpec for Testing

Reply:
fixed in [https://github.com/upb-uc4/University-Credits-4.0/pull/121/commits/710923f1f36afa50fc8faff88e18af10d9ceb896](https://github.com/upb-uc4/University-Credits-4.0/pull/121/commits/710923f1f36afa50fc8faff88e18af10d9ceb896)